src/mame/drivers/ladybug.cpp

```diff
@@ -135,42 +143,82 @@ void dorodon_state::decrypted_opcodes_map(address_map &map)
 }
 
 
-void sraider_state::sraider_cpu1_map(address_map &map)
+void mrsdyna_state::mrsdyna_cpu1_map(address_map &map)
 {
-	map(0x0000, 0x5fff).rom();
-	map(0x6000, 0x6fff).ram();
-	map(0x7000, 0x73ff).w("video", FUNC(ladybug_video_device::spr_w));
-	map(0x8005, 0x8005).r(FUNC(sraider_state::sraider_8005_r));  // protection check?
-	map(0x8006, 0x8006).writeonly().share("sound_low");
-	map(0x8007, 0x8007).writeonly().share("sound_high");
-	map(0x9000, 0x9000).portr("IN0");
-	map(0x9001, 0x9001).portr("IN1");
-	map(0x9002, 0x9002).portr("DSW0");
-	map(0x9003, 0x9003).portr("DSW1");
-	map(0xd000, 0xd7ff).w("video", FUNC(ladybug_video_device::bg_w));
-	map(0xe000, 0xe000).nopw();  //unknown 0x10 when in attract, 0x20 when coined/playing
+	// LS138 @ J4
+	map(0x0000, 0x1fff).rom();	// 2764 at R4
+	map(0x2000, 0x3fff).rom();	// 2764 at N4
+	map(0x4000, 0x5fff).rom();	// 2764 at M4
+	// LS138 @ J4 and LS139 @ H4
+	map(0x6000, 0x67ff).ram();  // 6116 @ K3, also clk on PAL K2 (16R6, U001)
+	map(0x6800, 0x6fff).ram();  // 6116 @ M3
+	map(0x7000, 0x73ff).w("video", FUNC(ladybug_video_device::spr_w)); // pin 29 on ribbon 
+	//map(0x77ff, 0x7fff);	// LS139 @ H4 pin7 is NC
+
+	// LS138 @ J4, Pin11 (0x8000-0x9fff) and
+	// LS138 @ N3 (bottom 3 bits)
+	// (all of these are read/write)
+	map(0x8005, 0x8005).mirror(0x1ff8).r(FUNC(mrsdyna_state::mrsdyna_protection_r));  // OE on PAL @ K2 (16R6, U001) (100x xxxx xxxx x101)
+	map(0x8006, 0x8006).mirror(0x1ff8).writeonly().share("sound_low");  // LS374 @ P6
+	map(0x8007, 0x8007).mirror(0x1ff8).writeonly().share("sound_high"); // LS374 @ R6
```

Comment:
Would it be better to use `latch8_device` for these so you get an automatic synch barrier on write?